reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/RepeatUntil.kt

```diff
@@ -0,0 +1,39 @@
+package com.badoo.reaktive.observable
+
+import com.badoo.reaktive.base.ErrorCallback
+import com.badoo.reaktive.base.ValueCallback
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.utils.serializer.serializer
+
+fun <T> Observable<T>.repeatUntil(predicate: () -> Boolean): Observable<T> =
+    observable { emitter ->
+        val observer =
+            object : ObservableObserver<T>, ValueCallback<T> by emitter, ErrorCallback by emitter {
+                // Prevents recursive subscriptions
+                private val serializer =
+                    serializer<Unit> {
+                        subscribe(this)
+                        true
+                    }
+
+                override fun onSubscribe(disposable: Disposable) {
+                    emitter.setDisposable(disposable)
+                }
+
+                override fun onComplete() {
+                    if (!emitter.isDisposed) {
+                        if (predicate()) {
```

Comment:
We should try-catch any callback invocations. Write something like:
```
emitter.tryCatch(block = { predicate }) {
    if (it) {
        complete
    } else {
        subscribe
    }
}
```

Reply:
Also please check `emitter.isDisposed` after predicate invocation (inside tryCatch callback). Because callback might be blocking.

Reply:
And a test for exception in predicate would be nice.

Reply:
Done